### ISSUE TITLE
TUEV Reproduction: ~22% BAC vs 62% Claim — Implementation Details & Questions

### ISSUE THREAD

**Issue by The-Obstacle-Is-The-Way** [BINE022/EEGPT#24]
## Summary
Despite matching the reference pipeline for TUEV event classification, our best eval Balanced Accuracy is ~22% (paper: 62.32%). We're seeking clarification on data version, preprocessing, and any balancing/normalization steps.

## Environment & Data

- **Dataset**: TUEV v2.0.1 (official train/eval folders)
- **Splits**: official directories (subject leakage avoided)
- **Preprocessing**:
  - 0.1–75 Hz bandpass + 50 Hz notch (MNE)
  - Resample to 200 Hz
  - Referential channels only (23), ordered, zero-padded if missing
  - Event extraction: 5 s (1000 samples) using triple-buffer around annotations with enforced 1 s event span
- **Storage**: cached segments (23,1000) in Volts (SI units)

## Model Configuration

- **Channel mapper (23→20)**:
  - Conv2dWithConstraint(23→20, kernel=1, bias=True) → BatchNorm2d → GELU
  - Depthwise Conv2d(20→20, kernel=(1,55), groups=20, padding=(0,27), bias=False) → BatchNorm2d → Dropout(0.8)
- **EEGPT backbone**:
  - channels=20 (FP1, FPZ, FP2, F7, F3, FZ, F4, F8, T7, C3, CZ, C4, T8, P7, P3, PZ, P4, P8, O1, O2)
  - time_steps=1000, patch_stride=64
  - return_all_temporal=True → features shape (B,15,4,512), flattened to 30,720
  - DropPath=0.0
- **Classifier**: Dropout(0.8) → LinearWithConstraint(30720→6, max_norm=1)

## Input Scaling

- Dataset in Volts; model forward scales to μV/100 (V→μV→÷100) before mapper
- EEGPT normalization disabled to match the reference magnitude

## Training Hyperparameters

- Optimizer: AdamW (lr=5e-4, eps=1e-8, betas=(0.9,0.999)), weight_decay=0.05
- LR schedule: cosine with warmup=5, min_lr=1e-6, layer_decay=0.65
- Loss: LabelSmoothingCrossEntropy(0.1)
- Mixed precision (AMP with GradScaler)
- Effective batch size ≈400 (accumulation)

## Results

- **Best eval BAC**: ~0.22 (22%) over 30 epochs
- **Pattern**: strong background recall, partial GPED, near-zero for rare classes

### Class Distribution (from our cache)

Training (4213 segments):
- bckg: 2121 (50.3%)
- gped: 880 (20.9%)
- artf: 489 (11.6%)
- pled: 463 (11.0%)
- eyem: 238 (5.6%)
- spsw: 22 (0.5%)

Eval (1471 segments):
- bckg: 800 (54.4%)
- gped: 374 (25.4%)
- artf: 124 (8.4%)
- eyem: 75 (5.1%)
- pled: 74 (5.0%)
- spsw: 24 (1.6%)

## Reproduction Command

```bash
uv run python experiments/eegpt_linear_probe/train_tuev_events.py \
  --data_dir data/datasets/tuev \
  --cache_dir data/datasets/tuev/cache \
  --eegpt_checkpoint data/models/pretrained/eegpt_mcae_58chs_4s_large4E.ckpt \
  --epochs 30 --batch_size 64 --num_workers 4
```

## Questions

1. **Data/version**: Which TUEV version (v2.0.0 vs v2.0.1)? Any file/layout differences?
2. **Preprocessing**: Authors' event extractor directly or reimplemented? Any differences in channel handling or event alignment?
3. **Balancing**: Class weights, balanced samplers, or oversampling/augmentations for rare classes?
4. **Scaling/Normalization**: Was μV/100 the intended magnitude? Any per-recording z-score normalization pre-EEGPT?
5. **Seeds/Runs**: Paper reports mean±std — how many runs/seeds?
6. **Replication ceiling**: Others report ~58% BAC; insights on closing the gap to 62%?

We can provide logs (confusion matrices, per-class recall), training configs, and code excerpts upon request. Thanks for helping the community reproduce these results.
